x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/queries/SemanticMatchQueryRewriteInterceptor.java

```diff
@@ -34,12 +33,8 @@ public class SemanticMatchQueryRewriteInterceptor implements QueryRewriteInterce
     public SemanticMatchQueryRewriteInterceptor() {}
 
     @Override
-    public QueryBuilder rewrite(QueryRewriteContext context, QueryBuilder queryBuilder) {
-
-        if (queryBuilder instanceof MatchQueryBuilder == false) {
-            return queryBuilder;
-        }
-
+    public QueryBuilder interceptAndRewrite(QueryRewriteContext context, QueryBuilder queryBuilder) {
+        assert (queryBuilder instanceof MatchQueryBuilder);
```

Comment:
Dumb question: how does this interceptor know to only intercept match queries?

Update: After reviewing the code further, I understand how this interceptor only intercepts `match` queries. However, this behavior is based on an implementation detail that I think could easily trip us up later. We use `QueryRewriteInterceptor#multi` to create an interceptor lookup map. However, `QueryRewriteContext` will accept _any_ `QueryRewriteInterceptor`, including one not created by `QueryRewriteInterceptor#multi`. Could we update the classes and method signatures to to enforce that that the rewrite context should take a map of interceptors?

We don't need to do it in this PR, but I think it's something good for a follow-up.

Reply:
I get it. I think that we certainly could do that, and I'd argue that if we decide to go down that route, maybe we get rid of the `CompositeQueryRewriteInterceptor` concept at all in favor of a service class that will do the right thing. I'll think about it, but I'd like to do that in a followup if we decide that's the best direction to go in, as it won't impact transport serialization or other features.